src/config-file.c

```diff
@@ -41,14 +41,16 @@ void config_file_free(struct config *config)
         g_free(config->auth_token);
         g_free(config->gateway_token);
         g_free(config->bundle_download_location);
+        g_free(config->client_cert);
+        g_free(config->client_key);
         g_hash_table_destroy(config->device);
 }
 
 static gboolean get_key_string(GKeyFile *key_file, const gchar* group, const gchar* key, gchar** value, const gchar* default_value, GError **error)
 {
         gchar *val = NULL;
         val = g_key_file_get_string(key_file, group, key, NULL);
-        if (val == NULL) {
+        if (val == NULL || !g_strcmp0(val, "")) {
```

Comment:
g_strcmp0() returns an integer, thus I would perfer using it as `g_strcmp0(val, "") == 0` as otherwise, the first guess could be that we check for inequality.

Another more readable option would be to use `g_str_equals()` as we ensured val is != NULL already.